core/src/main/java/org/aerogear/mobile/core/http/OkHttpResponse.java

```diff
@@ -153,8 +61,23 @@ public String stringBody() {
         }
     }
 
+    @Override
+    public InputStream streamBody() {
+        if (response != null) {
+            return response.body().byteStream();
+        } else {
+            return new ByteArrayInputStream(new byte[0]);
```

Comment:
Wouldn't be better to cache the empty response to avoid useless objects instantiations:
```java
private static final ByteArrayInputStream EMPTY_RESPONSE =new ByteArrayInputStream(new byte[0]);
...
...
    if (response != null) {
        return response.body().byteStream();
    } else {
        return EMPTY_RESPONSE;
```

Reply:
+1